### PR TITLE
Add common issues to the bug report template

### LINKED ISSUE #6298: Create Issue Templates
It may be time for us to come up with an issue template.

### PR DESCRIPTION
**Pull request by lnicola**
Closes #6298

### CODE REVIEW

**Comment by kjeremy** [comment]
I think this would close #6298

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -11,5 +11,16 @@ assignees: ''
...
-Please try to provide information which will help us to fix the issue faster. Minimal reproducible examples with few dependencies are especially lovely <3
+Before submitting, please make that you're not running into one of these known issues:
+
+ 1. local imports (`use` statements) don't work: #1165
+ 2. local items don't work: #1559
+ 3. the `rust-lang.rust` (official extension) conflicts with `rust-analyzer`
+ 4. some platform-specific imports are not resolved: #6038
+ 5. some settings are required for procedural macro and build script support (`rust-analyzer.cargo.loadOutDirsFromCheck`, `rust-analyzer.procMacro.enable`): #6448
+ 6. on-the-fly diagnostics are mostly unimplemented (`cargo check` diagnostics will be shown when saving a file)
+ 7. some types are not resolved in trait-heavy code
```
Comment (kjeremy) [inline .github/ISSUE_TEMPLATE/bug_report.md:22]: Is there a linked issue for this?

Comment (lnicola) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Probably more than one, added a random one.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -11,5 +11,16 @@ assignees: ''
 Troubleshooting guide: https://rust-analyzer.github.io/manual.html#troubleshooting
 Forum for questions: https://users.rust-lang.org/c/ide/14
 
-Please try to provide information which will help us to fix the issue faster. Minimal reproducible examples with few dependencies are especially lovely <3
+Before submitting, please make that you're not running into one of these known issues:
+
+ 1. local imports (`use` statements) don't work: #1165
+ 2. local items don't work: #1559
+ 3. the `rust-lang.rust` (official extension) conflicts with `rust-analyzer`: #6463
```
Comment (bjorn3) [inline .github/ISSUE_TEMPLATE/bug_report.md:18]:
```suggestion
 3. the official Rust extension (`rust-lang.rust`) conflicts with `rust-analyzer`: #6463
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -11,5 +11,16 @@ assignees: ''
 Troubleshooting guide: https://rust-analyzer.github.io/manual.html#troubleshooting
 Forum for questions: https://users.rust-lang.org/c/ide/14
 
-Please try to provide information which will help us to fix the issue faster. Minimal reproducible examples with few dependencies are especially lovely <3
+Before submitting, please make that you're not running into one of these known issues:
```
Comment (Veykril) [inline .github/ISSUE_TEMPLATE/bug_report.md:14]:
```suggestion
Before submitting, please make sure that you're not running into one of these known issues:
```

**Comment by flodiebold** [comment]
I'd leave out 7 and 8, these aren't so clear-cut and I don't get the impression that we get a lot of duplicates for them.

**Comment by kjeremy** [comment]
LGTM

**Comment by lnicola** [comment]
> I don't know whether having the extension incompatibility in there helps, especially since it's not immediately clear what kinds of problems this matches

We've seen #6461 and cases like this were also discussed in the `vscode-rust` repo, e.g. https://github.com/rust-lang/vscode-rust/issues/852. I think it's safe to say that they're incompatible.

**Comment by flodiebold** [comment]
I agree that they're incompatible, I'm just not sure that's worth including here. Especially if we can instead warn about it programmatically (#6463).

**Comment by lnicola** [comment]
Let's leave it in here until #6463 is implemented, then we can remove it. I reordered them a little so it ended up at the end of the list. Even if it's not a common issue (though I tend to disagree based on various reports I've seen), it won't hurt anyone.

**Review by flodiebold — APPROVED** [review]

**Comment by lnicola** [comment]
bors r=flodiebold

**Pull request merged**